From fcff8f6eadc5239d78a8758a9450a4361cd2b77c Mon Sep 17 00:00:00 2001
From: Ley Foon Tan <ley.foon.tan@intel.com>
Date: Fri, 14 Jun 2019 16:57:08 +0800
Subject: [PATCH 2/2] arm64: dts: stratix10: Add dma-ranges

Hardware can only do DMA to lower 2GB.

Add dma-ranges to limit it to 2GB.

Signed-off-by: Ley Foon Tan <ley.foon.tan@intel.com>
[This patch is from the Intel tech support directly]
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/altera/socfpga_stratix10.dtsi | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm64/boot/dts/altera/socfpga_stratix10.dtsi b/arch/arm64/boot/dts/altera/socfpga_stratix10.dtsi
index 9a9f31963441..02af470883db 100644
--- a/arch/arm64/boot/dts/altera/socfpga_stratix10.dtsi
+++ b/arch/arm64/boot/dts/altera/socfpga_stratix10.dtsi
@@ -94,6 +94,7 @@
 		device_type = "soc";
 		interrupt-parent = <&intc>;
 		ranges = <0 0 0 0xffffffff>;
+		dma-ranges = <0x00000000 0x00000000 0x00000000 0x80000000>;
 
 		base_fpga_region {
 			#address-cells = <0x1>;
-- 
2.17.1

